packages/client/src/authentication/Authentication.ts

```diff
@@ -35,6 +35,11 @@ export class Authentication implements IAuthentication {
     this.credentials = new CredentialsStorage(context.storage, context.environment.name);
   }
 
+  async fromRefreshToken(refreshToken: string): Promise<void> {
+    const credentials = new Credentials(undefined, refreshToken);
+    await this.credentials.set(credentials);
```

Comment:
@cesarenaldi I made it as simple as this.
shall we validate if the input is a valid jwt token and if it is a lens refresh token or leave it out to the consumer? if not valid then the API will fail on the refresh step anyway and it will happen the first time you call any client method. wdyt?

Reply:
I think a simple approach works for now. I have a question about the design choice.

Why you went for a method compared to a configuration option in the constructor?

Reply:
first that `client.authentication` module contains all methods related to authentication, so it's easier to find it here if you are checking what is possible, 
2nd, you might have a client instance from another place of your app and only authenticate later when you have a token.

Reply:
it's probably not clear but each form of authentication will overwrite the previous one

Reply:
I see and understand the rationale. The only thing is that the method name might not be intelligible. the "fromSomething" kinda make sense as static factory method (e.g. `Car.fromParts(...)`) but not sure if works well as instance method of a submodule.

Reply:
I'm open to better ideas 😄

Reply:
I renamed it to

```ts
/**
* Authenticate with an existing, valid refresh token.
*/
authenticateWith({ refreshToken }: { refreshToken: string }): Promise<void>;
```